app/controllers/admin/videos_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Admin::VideosController < ApplicationController
+
+  before_action :authorize
+  before_action :require_admin
```

Comment:
This and the related method can be moved to AdminsController and this controller can inherit from there. This will allow you to organize the admin requirement logic in one place, by itself.